### PR TITLE
checker: use CheckerSchedulerType for checker 

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/8379

### What is changed and how does it work?

```commit-message
checker: use CheckerSchedulerType for checker
- move metrics to proper place
- rename GetType() to Name()
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by okJiang** [comment]
Could you take a look? @JmPotato @nolouch @rleungx

File: pkg/schedule/config/type.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+type CheckerSchedulerType string
```
Comment (rleungx) [inline pkg/schedule/config/type.go:17]: Why not put it under the checker package?

Comment (okJiang) [reply pkg/schedule/config/type.go:17]: Because the scheduler will also be placed here later.

Comment (rleungx) [reply pkg/schedule/config/type.go:17]: Then, how about just using type as the package name?

Comment (okJiang) [reply pkg/schedule/config/type.go:17]: sgtm

**Review by rleungx — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

**Pull request merged**